website/project/model.py

```diff
@@ -2974,13 +2986,20 @@ def to_json(self):
 
 
 class Sanction(StoredObject):
-    """Sanction object is a generic way to track approval states"""
-
+    """Sanction class is a generic way to track approval states"""
+    # Tell modularodm not to attach backends
     abstract = True
 
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    # Neither approved not cancelled
     UNAPPROVED = 'unapproved'
+    # Has approval
     APPROVED = 'approved'
+    # Rejected by at least one person
     REJECTED = 'rejected'
+    # One of 'unapproved', 'active', 'cancelled', or 'completed
+    state = fields.StringField(default=UNAPPROVED)
```

Comment:
That's true, but we don't know how this class will be used in the future, and validation is an easy safeguard. (it's easier if it was there from the start) We can revisit this with the help of https://github.com/CenterForOpenScience/osf.io/pull/4343 or similar.